### PR TITLE
Create Graph Coloring kata, part 1: vertex coloring problem

### PR DESCRIPTION
**Pull request by tcNickolas**
I skipped the register initialization task from the original kata, since it was not used in the complete solution, and merged two other intro tasks into their similar, more challenging variants to reduce the total number of tasks.
The next PR will finish the kata, adding part 2 "weak coloring problem" and the e2e demo of Grover's algorithm.

### CODE REVIEW

File: katas/content/solving_graph_coloring/Common.qs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    ) : Bool {
+        // Message($"Testing V = {V}, edges = {edges}");
+        let N = 2 * V;
+        use (coloringRegister, target) = (Qubit[N], Qubit());
+        // Try all possible colorings of 4 colors on V vertices and check if they are calculated correctly.
+        // Hack: fix the color of the first vertex, since all colorings are agnostic to the specific colors used.
+        for k in 0 .. (1 <<< (N - 2)) - 1 {
+            // Prepare k-th coloring
+            let binary = [false, false] + IntAsBoolArray(k, N - 2);
+            ApplyPauliFromBitString(PauliX, true, binary, coloringRegister);
```
Comment (cesarzc) [inline katas/content/solving_graph_coloring/Common.qs:58]: Would using a `within apply` construct make sense here?

File: katas/content/solving_graph_coloring/read_coloring/Verification.qs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    @EntryPoint()
+    operation CheckSolution() : Bool {
+        for nBits in 1..3 {
+            for V in 1..3 {
+                use qs = Qubit[V * nBits];
+                for state in 0..(1 <<< (V * nBits)) - 1 {
+                    // Prepare the register in the input state
+                    let binaryState = IntAsBoolArray(state, V * nBits);
+                    ApplyPauliFromBitString(PauliX, true, binaryState, qs);
```
Comment (cesarzc) [inline katas/content/solving_graph_coloring/read_coloring/Verification.qs:15]: Would a `within apply` construct make sense here too?

File: katas/content/solving_graph_coloring/index.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    "path": "./vertex_coloring/"
+})
+
+
+@[section]({
+    "id": "solving_graph_coloring__weak_coloring",
+    "title": "Weak Coloring Problem"
+})
+
+TBD
```
Comment (cesarzc) [inline katas/content/solving_graph_coloring/index.md:67]: Is this meant to be here? Maybe a "ToDo" with some context would be more appropriate.

File: katas/content/solving_graph_coloring/index.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    "id": "solving_graph_coloring__using_grover",
+    "title": "Using Grover's Algorithm to Solve Graph Coloring Problems"
+})
+
+In this lesson, you will experiment with using Grover's algorithm to solve graph coloring problems.
+
+Notice that in this case, it's not as easy to know the number of solutions to the problem upfront as it was for the prefix function used in the "Grover's Search Algorithm" kata.
+Experiment with choosing the number of iterations at random. How does this affect the success probability?
+
+TBD
```
Comment (cesarzc) [inline katas/content/solving_graph_coloring/index.md:80]: Similarly to my other comment, maybe a "ToDo" with more context would be helpful.

**Review by cesarzc — APPROVED** [review]

File: katas/content/solving_graph_coloring/Common.qs (hunk 1 above)
Comment (tcNickolas) [reply katas/content/solving_graph_coloring/Common.qs:58]: This is pretty much the only scenario where I lean towards not using it. The part that needs to be uncomputed is very short, one line, and simple, so using it doesn't shorten the code, only increase the number of nested brackets

File: katas/content/solving_graph_coloring/read_coloring/Verification.qs (hunk 2 above)
Comment (tcNickolas) [reply katas/content/solving_graph_coloring/read_coloring/Verification.qs:15]: Same as above, for very simple apply code doesn't improve the code much

File: katas/content/solving_graph_coloring/color_equality/solution.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+The goal is to flip the qubit $\ket{y}$ if and only if each of the matching pairs of qubits in the arrays `x0` and `x1` are in the same state.
+
+You can check whether two qubits are in the same state by computing their $\textrm{XOR}$: if their state was the same, their $\textrm{XOR}$ will be $0$. You can use the $\textrm{CNOT}$ gate to compute $\textrm{XOR}$ of two qubits in place, for example, using the qubit of the array `x0` as the control and the qubit of the array `x1` as the target.
+
+Once you've used $nBits$ $\textrm{CNOT}$ gates to compute all pairwise $\textrm{XOR}$s, you'll need to flip the target qubit $\ket{y}$ only if all  qubits in `x1` are in the $\ket{0}$ state.
```
[the remark this answers is not in the document]
Comment (ScottCarda-MS) [reply katas/content/solving_graph_coloring/color_equality/solution.md:5]: I've confirmed that this is also an issue on the playground.

**Review by ScottCarda-MS — APPROVED** [review]

File: katas/content/solving_graph_coloring/color_equality/solution.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (tcNickolas) [reply katas/content/solving_graph_coloring/color_equality/solution.md:5]: Interesting, MathJax doesn't mind (but renders `x1` after it oddly)
image
But the "main" playground does mind
image
The 's trick fixed both, thank you for you vigilance!

File: katas/content/solving_graph_coloring/vertex_coloring/index.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+1. The number of vertices in the graph $V$ ($V \leq 6$).
+2. An array of $E$ tuples of integers, representing the edges of the graph ($E \leq 12$).
+Each tuple gives the indices of the start and the end vertices of the edge.
+The vertices are numbered $0$ through $V - 1$.
+3. An array of $2V$ qubits in an arbitrary state $\ket{x}$ representing the assigned coloring of the vertices using four colors ($nBits = 2$) in the same format as in the exercise "Read Coloring From a Qubit Array".
+4. A qubit in an arbitrary state $\ket{y}$ (output/target qubit).
+
+For example, the graph `0 -- 1 -- 2` has $V = 3$ and `edges = [(0, 1), (1, 2)]`. 
+The color assignments for this graph are represented with $6$ qubits; some of the valid colorings are $\ket{000110}$ and $\ket{001100}$.
```
[the remark this answers is not in the document]
Comment (tcNickolas) [reply katas/content/solving_graph_coloring/vertex_coloring/index.md:11]: Good point! I refer to an earlier task in the input description, but I don't have an example there; I expanded both task descriptions

**Pull request merged**